DuckDuckGo/LocalNotificationsLogic.swift

```diff
@@ -0,0 +1,256 @@
+    
+    private func cancelPrivacyNotification() {
+        LocalNotifications.shared.cancelNotifications(withIdentifiers: [Notification.privacy.identifier])
+        store.didCancel(notification: .privacy)
+    }
+    
+    func willLeaveApplication() {
+        
+        if variantManager.isSupported(feature: .dayZeroNotification),
+            store.notificationsEnabled,
+            store.scheduleStatus(for: .privacy) == nil {
+            schedulePrivacyNotification()
+        }
+        
+        if variantManager.isSupported(feature: .dayOneNotification),
+            store.notificationsEnabled,
+            store.scheduleStatus(for: .homeRow) == nil {
+            scheduleHomeRowNotification()
+        }
+    }
+        
+    private func schedulePrivacyNotification() {
+        let privacyData = PrivacyReportDataSource()
+        
+        let title: String
+        let body: String
+        if privacyData.trackersCount < Constants.privacyNotificationInfoTreshold {
+            title = "Ready to take back your privacy?"
+            body = "Keep browsing with DuckDuckGo to block trackers and encrypt connections."
+        } else {
+            title = "Success! The internet just got less creepy."
+            body = "While using DuckDuckGo you blocked \(privacyData.trackersCount) trackers and secured \(privacyData.httpsUpgradesCount) unencrypted connections."
+        }
+        
+        LocalNotifications.shared.scheduleNotification(title: title,
+                                                       body: body,
+                                                       identifier: Notification.privacy.identifier,
+                                                       timeInterval: Constants.privacyNotificationDelay)
+        store.didSchedule(notification: .privacy, date: Date().addingTimeInterval(Constants.privacyNotificationDelay))
+    }
+    
+    func fireDateForHomeRowNotification(currentDate: Date = Date()) -> (DateComponents, Date)? {
+        var components = Calendar.current.dateComponents(in: .current, from: currentDate)
+        
+        if let hour = components.hour, hour > 3 {
+            components.day = (components.day ?? 0) + 1
+        }
+        
+        components.hour = 10
+        components.minute = 0
+        components.second = 0
+
+        let earliestDate = currentDate.addingTimeInterval(12 * 60 * 60)
+        
+        guard var date = Calendar.current.date(from: components) else { return nil }
+        
+        if date < earliestDate {
+            components = Calendar.current.dateComponents(in: .current, from: earliestDate)
+            date = earliestDate
+        }
+        
+        return (components, date)
+    }
+    
+    private func scheduleHomeRowNotification() {
+        
+        if let (_, date) = fireDateForHomeRowNotification() {
+            let title = "Prioritize your privacy"
+            let body = "Move the DuckDuckGo app to your home dock for easy access to private browsing."
```

Comment:
Is "home dock" the term finally decided on? I thought I read somewhere "home row" ?

Reply:
There was a bit of discussion around that an even PickFu tests - everything is described here: https://app.asana.com/0/856498667320406/1134638273464225